File: dota-zone/dotazone/src/main/java/br/com/dotazone/view/activity/TabActivity.java
package br.com.dotazone.view.activity;

import android.os.Bundle;
import androidx.viewpager.widget.ViewPager;
import androidx.viewpager.widget.ViewPager.OnPageChangeListener;
import androidx.drawerlayout.widget.DrawerLayout;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.ImageView;
import android.widget.RelativeLayout;

import com.google.android.gms.ads.AdView;
import com.google.android.gms.analytics.GoogleAnalytics;
import com.google.android.material.tabs.TabLayout;

import br.com.dotazone.DotaZoneApplication;
import br.com.dotazone.DotaZoneBrain;
import br.com.dotazone.DotazoneMenu;
import br.com.dotazone.R;
import br.com.dotazone.model.entity.AdMobBanner;
import br.com.dotazone.view.adapter.HeroAdapter;
import br.com.dotazone.view.adapter.ItemAdapter;

public class TabActivity extends BaseActivity implements OnPageChangeListener, OnClickListener {

    public static final int TAB_ITEM = 1;
    public static final int TAB_HERO = 0;
    public static boolean isBuild;
    public DotazoneMenu mMenu;
    private AdView adView;
    private DrawerLayout mDrawerLayout;
    private RelativeLayout mDrawerList;
    private ImageView mImageLogo;
    private ImageView mImageMenu;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.tab_view);

        ((DotaZoneApplication) getApplication()).getTracker(DotaZoneApplication.TrackerName.APP_TRACKER);

        initComponents();

    }

    @Override
    protected void onStart() {
        super.onStart();

        GoogleAnalytics.getInstance(this).reportActivityStart(this);
    }

    @Override
    protected void onStop() {
        super.onStop();

        // Stop the analytics tracking
        GoogleAnalytics.getInstance(this).reportActivityStop(this);
    }

    @Override
    public void onPageScrollStateChanged(int state) {
    }

    @Override
    public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {

    }

    @Override
    public void onPageSelected(int position) {
    }

    @Override
    public void onPause() {
        adView.pause();
        super.onPause();
    }

    @Override
    public void onResume() {
        super.onResume();
        adView.resume();

        int id = getIntent().getExtras().getInt(MainActivity.class.getName());
        if (getIntent() != null && id == DotazoneMenu.MENU_HERO) {
            mMenu.checkHeroMenu();

        } else if (id == DotazoneMenu.MENU_ITEM) {
            mMenu.checkItemMenu();
        } else {

            mMenu.checkBuildMenu();
        }
    }

    @Override
    public void onDestroy() {
        adView.destroy();
        super.onDestroy();
    }

    private void setAdapterPage(ViewPager pager) {

        ItemAdapter itemAdapter = new ItemAdapter(getSupportFragmentManager(), getApplicationContext());
        HeroAdapter heroAdapter = new HeroAdapter(getSupportFragmentManager(), this);

        int id = getIntent().getExtras().getInt(MainActivity.class.getName());

        if (getIntent() != null && id == DotazoneMenu.MENU_HERO) {
            pager.setAdapter(heroAdapter);
            mMenu.checkHeroMenu();

        } else if (id == DotazoneMenu.MENU_ITEM) {
            pager.setAdapter(itemAdapter);
            mMenu.checkItemMenu();
        } else {
            pager.setAdapter(heroAdapter);
            mMenu.checkBuildMenu();
        }
    }

    @Override
    public void initComponents() {

        mDrawerLayout = (DrawerLayout) findViewById(R.id.drawer_layout);
        mDrawerList = (RelativeLayout) findViewById(R.id.list_slidermenu);
        mImageLogo = (ImageView) findViewById(R.id.header_imageLogo);
        mImageMenu = (ImageView) findViewById(R.id.header_imageMenu);
        mImageLogo.setOnClickListener(this);
        mImageMenu.setOnClickListener(this);
        mMenu = new DotazoneMenu(this, mDrawerLayout, mDrawerList);

        ViewPager pager = (ViewPager) findViewById(R.id.tab_view_pager);
        adView = (AdView) findViewById(R.id.tab_view_admob);
        setAdapterPage(pager);
        TabLayout indicator = (TabLayout) findViewById(R.id.indicator);
        indicator.setupWithViewPager(pager);
        //indicator.setOnPageChangeListener(this);
        new AdMobBanner().createBanner(this, adView, DotaZoneBrain.isPremium);

    }

    @Override
    public void setActionErrorOk() {
        finish();
    }

    @Override
    public void setActionErrorCancel() {
        finish();

    }

    @Override
    public void onClick(View v) {

        mDrawerLayout.openDrawer(mDrawerList);

    }

    @Override
    public void finish() {
        super.finish();
        mMenu.defaultMenu();
    }
}

File: dota-zone/app/src/main/java/com/br/dotazone/view/activity/TabActivity.kt
package com.br.dotazone.view.activity

import android.os.Bundle
import android.view.View
import android.widget.ImageView
import android.widget.RelativeLayout
import androidx.drawerlayout.widget.DrawerLayout
import androidx.viewpager.widget.ViewPager
import androidx.viewpager.widget.ViewPager.OnPageChangeListener
import com.br.dotazone.DotaZoneApplication
import com.br.dotazone.DotaZoneBrain.isPremium
import com.br.dotazone.R
import com.br.dotazone.model.entity.AdMobBanner
import com.br.dotazone.view.adapter.HeroAdapter
import com.br.dotazone.view.adapter.ItemAdapter
import com.com.dotazone.DotazoneMenu
import com.google.android.gms.ads.AdView
import com.google.android.gms.analytics.GoogleAnalytics
import com.google.android.material.tabs.TabLayout


class TabActivity : BaseActivity(), OnPageChangeListener, View.OnClickListener {
	var mMenu: DotazoneMenu? = null
	private var adView: AdView? = null
	private var mDrawerLayout: DrawerLayout? = null
	private var mDrawerList: RelativeLayout? = null
	private var mImageLogo: ImageView? = null
	private var mImageMenu: ImageView? = null
	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		setContentView(R.layout.tab_view)
		(application as DotaZoneApplication).getTracker(DotaZoneApplication.TrackerName.APP_TRACKER)
		initComponents()
	}

	override fun onStart() {
		super.onStart()
		GoogleAnalytics.getInstance(this).reportActivityStart(this)
	}

	override fun onStop() {
		super.onStop()

		// Stop the analytics tracking
		GoogleAnalytics.getInstance(this).reportActivityStop(this)
	}

	override fun onPageScrollStateChanged(state: Int) {}
	override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {}
	override fun onPageSelected(position: Int) {}
	public override fun onPause() {
		adView!!.pause()
		super.onPause()
	}

	public override fun onResume() {
		super.onResume()
		adView!!.resume()
		val id = intent.extras!!.getInt(MainActivity::class.java.name)
		if (intent != null && id == DotazoneMenu.MENU_HERO) {
			mMenu!!.checkHeroMenu()
		} else if (id == DotazoneMenu.MENU_ITEM) {
			mMenu!!.checkItemMenu()
		} else {
			mMenu!!.checkBuildMenu()
		}
	}

	public override fun onDestroy() {
		adView!!.destroy()
		super.onDestroy()
	}

	private fun setAdapterPage(pager: ViewPager) {
		val itemAdapter = ItemAdapter(supportFragmentManager, applicationContext)
		val heroAdapter = HeroAdapter(supportFragmentManager, this)
		val id = intent.extras!!.getInt(MainActivity::class.java.name)
		if (intent != null && id == DotazoneMenu.MENU_HERO) {
			pager.adapter = heroAdapter
			mMenu!!.checkHeroMenu()
		} else if (id == DotazoneMenu.MENU_ITEM) {
			pager.adapter = itemAdapter
			mMenu!!.checkItemMenu()
		} else {
			pager.adapter = heroAdapter
			mMenu!!.checkBuildMenu()
		}
	}

	override fun initComponents() {
		mDrawerLayout = findViewById<View>(R.id.drawer_layout) as DrawerLayout
		mDrawerList = findViewById<View>(R.id.listSliderMenu) as RelativeLayout
		mImageLogo = findViewById<View>(R.id.header_imageLogo) as ImageView
		mImageMenu = findViewById<View>(R.id.header_imageMenu) as ImageView
		mImageLogo!!.setOnClickListener(this)
		mImageMenu!!.setOnClickListener(this)
		mMenu = DotazoneMenu(this, mDrawerLayout, mDrawerList)
		val pager = findViewById<View>(R.id.tab_view_pager) as ViewPager
		adView = findViewById<View>(R.id.tab_view_admob) as AdView
		setAdapterPage(pager)
		val indicator = findViewById<View>(R.id.indicator) as TabLayout
		indicator.setupWithViewPager(pager)
		//indicator.setOnPageChangeListener(this);
		AdMobBanner().createBanner(this, adView!!, isPremium)
	}

	override fun setActionErrorOk() {
		finish()
	}

	override fun setActionErrorCancel() {
		finish()
	}

	override fun onClick(v: View) {
		mDrawerLayout!!.openDrawer(mDrawerList!!)
	}

	override fun finish() {
		super.finish()
		mMenu!!.defaultMenu()
	}

	companion object {
		const val TAB_ITEM = 1
		const val TAB_HERO = 0
		var isBuild = false
	}
}
